src/main/java/seedu/address/logic/parser/Parser.java

```diff
@@ -128,12 +129,24 @@ private Command prepareAdd(String args){
     private Command prepareDelete(String args) {
 
         Optional<Integer> index = parseIndex(args);
+        // check if argument is index
+
         if(!index.isPresent()){
-            return new IncorrectCommand(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
+            String nameMatch = args.length() > 0 ? args.substring(1) : args;
+            // remove white space from command delimiter
+
+            if(nameMatch.contentEquals("")) {
+                return new IncorrectCommand(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
+            }
+            // invalid delete command if no arguments
+
+            return new DeleteCommand(nameMatch);
+            // create delete by name
         }
 
         return new DeleteCommand(index.get());
+        // create delete by index
```

Comment:
Where are you checking if the index is present? What if neither name nor index is present?